packages/scanner/src/rules/lib/util.ts

```diff
@@ -15,6 +17,17 @@ export async function appmapDirFromConfig(): Promise<string | undefined> {
   }
 }
 
+export async function writeFileAtomically(fileName: string, data: any): Promise<void> {
+  const tmpDir = await mkdtemp(tmpdir() + '/');
+  const tempFileName = join(tmpDir, basename(fileName));
+  try {
+    await writeFile(tempFileName, data);
+    await rename(tempFileName, fileName);
+  } finally {
+    await rm(tmpDir, { recursive: true }).catch((err) => console.warn(err));
+  }
```

Comment:
This is incorrect — `tmp` is likely on a different volume than the target file and the rename will fail with `EXDEV` (cross-device link not permitted). The temp file needs to be in the target directory. (You could put it in temp then move, but this won't be atomic and it will needlessly copy data around.)

Consider using a library like [write-file-atomic](https://www.npmjs.com/package/write-file-atomic) instead of rolling your own, this stuff is notoriously finicky.